src/components/pages/blog-post/blog-post.tsx

```diff
@@ -33,114 +56,246 @@ const PanelContainer = styled.div`
   }
 `;
 
-const contentfulRenderOptions: Options = {
-  renderMark: {
-    [MARKS.CODE]: (text) => <code className="language-text">{text}</code>,
-  },
-  renderNode: {
-    [INLINES.HYPERLINK]: (props, children) =>
-      customComponents.a({
-        children,
-        href: props.data.uri,
-        target: '_blank',
-        rel: 'nofollow noopener noreferrer',
+/**
+ * Custom Contentful Rich Text Renderer
+ *
+ * Will automatically filter out ContentfulFootnote elements and appends an
+ * ordered list of footnotes after the rendered rich text.
+ *
+ * @param data rich text data
+ * @returns rendered elements
+ */
+const customContentfulRenderer = (
```

Comment:
Great refactoring to allow for the footnotes to be included. I think having an explicit renderer for contentful it's time to put it into a separate file and add a few tests to ensure the matching data is output (not in this PR but as some follow up or delegate to a working student for the learning effect)